lib/compute/jenkins-main-node.ts

```diff
@@ -0,0 +1,282 @@
+/**
```

Comment:
What happens to git history for the files that were moved? We should move the files in such a way that the history is preserved.

Reply:
We have a small amount of code and people working on this project, I don't think the cost is worthwhile.  The infra repository will maintain this history on its deleted files.

Reply:
Ohh, I didn't know we would have confidential information as a part of commit messages. Although , it is strange to have such info in commit messages, irrespective

Reply:
In a private branch it doesn't matter what info we have in there :) so we can go wild.  We just need to be really really careful when pulling stuff out